src/systems/optical_tactile_plugin/OpticalTactilePlugin.cc

```diff
@@ -170,547 +161,472 @@ class ignition::gazebo::systems::OpticalTactilePluginPrivate
+    // that all entities have been created when Configure is called
+    this->dataPtr->Load(_ecm);
+    this->dataPtr->initialized = true;
+  }
+
+  // This is not an "else" because "initialized" can be set
+  // in the if block above
+  if (this->dataPtr->initialized)
+  {
+    // Get the first object being touched by the sensor
+    // We assume there's only one object being touched
+    auto csd = _ecm.Component<components::ContactSensorData>(
+      this->dataPtr->sensorCollisionEntity);
+    for (const auto &contact : csd->Data().contact())
     {
-      this->cameraUpdateRate = depthCameraSdf->Get<float>("update_rate");
+      this->dataPtr->objectCollisionEntity =
+        contact.collision2().id();
+      break;
     }
 
+    // Get the Depth Camera pose
+    ignition::math::Pose3d sensorPose =
+      _ecm.Component<components::Pose>(
+        this->dataPtr->model.Entity())->Data();
+
     // Depth Camera data is float, so convert Pose3d to Pose3f
-    this->depthCameraOffset = ignition::math::Pose3f(
-        offset.Pos().X(), offset.Pos().Y(), offset.Pos().Z(),
-        offset.Rot().W(), offset.Rot().X(), offset.Rot().Y(),
-        offset.Rot().Z());
-
-    // Configure subscriber for Depth Camera images
-    std::string topic = "/depth_camera/points";
-    if (depthCameraSdf->HasElement("topic"))
-      topic = "/" + depthCameraSdf->Get<std::string>("topic") + "/points";
-
-    if (!this->node.Subscribe(topic,
-        &OpticalTactilePluginPrivate::DepthCameraCallback,
-        this))
-    {
-      ignerr << "Error subscribing to topic " << "[" << topic << "]. "
-        << "<topic> must not contain '/'" << std::endl;
-      return;
-    }
+    this->dataPtr->sensorWorldPose = ignition::math::Pose3f(
+      sensorPose.Pos().X(), sensorPose.Pos().Y(), sensorPose.Pos().Z(),
+      sensorPose.Rot().W(), sensorPose.Rot().X(), sensorPose.Rot().Y(),
+      sensorPose.Rot().Z());
+  }
 }
 
 //////////////////////////////////////////////////
-void OpticalTactilePluginPrivate::Enable(const bool _value)
+void OpticalTactilePlugin::PostUpdate(
+  const ignition::gazebo::UpdateInfo &_info,
+  const ignition::gazebo::EntityComponentManager &_ecm)
 {
-    // Just a placeholder method for the moment
-    _value;
-}
+  IGN_PROFILE("TouchPlugin::PostUpdate");
 
-//////////////////////////////////////////////////
-void OpticalTactilePluginPrivate::FilterOutCollisions(
-                            const EntityComponentManager &_ecm,
-                            const std::vector<Entity> &_entities)
-{
-    if (_entities.empty())
-        return;
+  // Nothing left to do if paused.
+  if (_info.paused)
+    return;
 
-    // Clear vector that contains old data
-    this->filteredCollisionEntities.clear();
+  auto* contacts =
+    _ecm.Component<components::ContactSensorData>(
+      this->dataPtr->sensorCollisionEntity);
 
-    // Get collisions from the sensor
-    for (const Entity &entity : _entities)
+  if (contacts)
```

Comment:
Do you say that because `InterpolateContactData` is not publishing the contacts?
Should I write a comment explaining why is it commented out?

Reply:
I think we said we don't need to interpolate anymore, because it is limited to a box shape, and we only did it as a placeholder before we had depth camera images, right? I don't remember if we decided something else. I think we only need to keep the actual contacts from the physics engine, and not the ones interpolated, because they can be very off if the contacts are on a mesh.

Reply:
Removed code: https://github.com/ignitionrobotics/ign-gazebo/pull/229#discussion_r469805612